bundler.d/nulldb.rb

```diff
@@ -0,0 +1 @@
+gem 'activerecord-nulldb-adapter'
```

Comment:
any reason to not use a group like we do for other `bundler.d` files?

Reply:
Seems activerecord loading does not like this:

```
vagrant@centos7-katello-devel-stable foreman (fixes-29567)$ bundle exec rake apipie:cache RAILS_ENV=production cache_part=resources DATABASE_URL=nulldb://nohost --trace
/home/vagrant/foreman/lib/foreman.rb:8: warning: already initialized constant Foreman::UUID_REGEXP
/home/vagrant/foreman/lib/foreman.rb:8: warning: previous definition of UUID_REGEXP was here
** Invoke apipie:cache (first_time)
** Invoke environment (first_time)
** Execute environment
Apipie cache enabled but not present yet. Run apipie:cache rake task to speed up API calls.
** Execute apipie:cache
2020-04-19 13:58:50 +0000 | Started
rake aborted!
NullDB not configured. Require a framework, ex 'nulldb/rails'
/home/vagrant/.gem/ruby/gems/activerecord-nulldb-adapter-0.4.0/lib/nulldb/core.rb:18:in `configuration'
/home/vagrant/.gem/ruby/gems/activerecord-nulldb-adapter-0.4.0/lib/active_record/connection_adapters/nulldb_adapter/core.rb:146:in `columns'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/connection_adapters/schema_cache.rb:69:in `columns'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/connection_adapters/schema_cache.rb:75:in `columns_hash'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/model_schema.rb:466:in `load_schema!'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/attributes.rb:234:in `load_schema!'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/attribute_decorators.rb:51:in `load_schema!'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/model_schema.rb:459:in `block in load_schema'
/opt/rh/rh-ruby25/root/usr/share/ruby/monitor.rb:226:in `mon_synchronize'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/model_schema.rb:456:in `load_schema'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/model_schema.rb:346:in `attribute_types'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/attribute_methods.rb:230:in `has_attribute?'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/relation/calculations.rb:196:in `block in pluck'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/relation/calculations.rb:195:in `map'
/home/vagrant/.gem/ruby/gems/activerecord-5.2.1/lib/active_record/relation/calculations.rb:195:in `pluck'
/home/vagrant/foreman/app/models/permission.rb:14:in `resources'
/home/vagrant/foreman/app/controllers/api/v2/template_inputs_controller.rb:41:in `block (2 levels) in <class:TemplateInputsController>'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:322:in `instance_exec'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:322:in `initialize'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:315:in `new'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:315:in `build'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:34:in `block in find'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:33:in `each'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/validator.rb:33:in `find'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:93:in `initialize'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:21:in `new'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/param_description.rb:21:in `from_dsl_data'
/home/vagrant/.gem/ruby/gems/apipie-rails-0.5.17/lib/apipie/method_description.rb:55:in `block in initialize'

```

Reply:
Would this mean users need to install the nulldb package for foreman to start in production?

Reply:
Nah, comes down to packaging. In the RPM environment the PR includes this file only in `foreman-plugin` sub-package which is used for builds only.

Reply:
I think that if `foreman-build` includes the file, it should be ok.